benchmarks/fasta/ruby/bench.rb

```diff
@@ -62,6 +63,7 @@ def make_random_fasta(table, n)
     prob = 0.0
     rwidth = (1..width)
     table.each{|v| v[1]= (prob += v[1])}
+    table = table.collect{|v| prob += v[1]; [v[0], prob] }
```

Comment:
yes, but ``prob`` is a local variable, so that's fine. the previous code was mutating ``table``, which is a global.